### PR TITLE
55 create a bootstrap reference transcript using the whisper models

### PR DESCRIPTION
**Pull request by marconoel**
Whisper is a general-purpose speech recognition model. It is trained on a large dataset of diverse audio and is also a multi-task model that can perform multilingual speech recognition as well as speech translation and language identification. - see here for more information https://openai.com/blog/whisper/

Some customers may have a lot of audio files to build their training and test sets, but they do not have time to transcribe them to have a valid reference.

This new Python module will allow users to build a pretty accurate reference using the "whisper" model.

DCO 1.1 Signed-off-by: Marco Noel - [marco.noel@ca.ibm.com](mailto:marco.noel@ca.ibm.com)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,3 +222,43 @@ Instructions for creating a directory structure for organizing input and output
...
+- default language
+
+You need to update the python script `reference-transcripts.py`, and update the "auto_detect" variable
+
+- auto_detect = True # Set to True if you do not know the language of the audio
+- default_lang = 'en' # If auto_detect is False, this value will be used for all audio files
+
+TIP: You can use the same first 2 letters of the Watson STT model you wish to evaluate (eg. en-GB_Telephony ==> "en" / pt-BR_Telephony ==> "pt")
+
+NOTE: The "whisper" language detection may not be accurate. In the CSV file, a column called "Language - CAN BE DELETED" is created as an informational indicator of the "whisper" language used (detected or by default).
```
Comment (andrewrfreed) [inline README.md:254]: I'd rather not include a forced manual step.  
Consider these options:
* Don't write the language
* Write 2 CSV, one with language and one without.

We are already logging the language in the print statement.

File: whisper-script/reference-transcripts.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+model = whisper.load_model("medium")
+
+#list audio files in current folder
+
+files=os.listdir('.')
+# Select the language of the Whisper model
+auto_detect = True # Set to True if you do not know the language of the audio
+default_lang = 'en' # If auto_detect is True, this value is ignored
+
+#Process each audio files in whisper, wirting in reference_transcriptions_whisper.csv
```
Comment (andrewrfreed) [inline whisper-script/reference-transcripts.py:13]: typo

File: whisper-script/reference-transcripts.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+with open('./reference_transcriptions_whisper.csv','w', encoding='utf-8') as csvfile:
+    #Writing CSV column headers
+    csvfile.write('Language - CAN BE DELETED,Audio File Name,Reference')
```
Comment (andrewrfreed) [inline whisper-script/reference-transcripts.py:16]: I'd prefer not to add a manual step if possible.  I trust your judgement.

File: whisper-script/reference-transcripts.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import whisper
```
Comment (andrewrfreed) [inline whisper-script/reference-transcripts.py:1]: Consider changing filename to something like `extra-reference-transcripts.py` or `whisper-transcription.py` or `extra-whisper-transcription.py` so the name differentiates itself further from the base/required tooling.

**Review by andrewrfreed — CHANGES_REQUESTED** [review]
Nice way to get a quick, independent bootstrap transcription.  Some suggestions added for making this tool easier to use.

File: README.md (hunk 1 above)
Comment (marconoel) [reply README.md:254]: working on it

File: whisper-script/reference-transcripts.py (hunk 2 above)
Comment (marconoel) [reply whisper-script/reference-transcripts.py:13]: working on it

File: whisper-script/reference-transcripts.py (hunk 3 above)
Comment (marconoel) [reply whisper-script/reference-transcripts.py:16]: agreed - working on it

File: whisper-script/reference-transcripts.py (hunk 4 above)
Comment (marconoel) [reply whisper-script/reference-transcripts.py:1]: agreed - working on it

**Comment by marconoel** [comment]
As this may cause confusion around this WER tool, I am closing this issue. May revisit later.

**Pull request closed without merging**